### PR TITLE
Update EIP-4494: Include EIP-1271 and EIP-2098 as required

### PR DESCRIPTION
**Pull request by TimDaub**
EIP-1271:
- EIP-4494 currently makes no mention of EIP-1271 but I strongly assume that there is interest to support signature validation through a smart contract
- There are strong reasons to believe so as [@dievardump's reference implementation](https://github.com/dievardump/erc721-with-permits/blob/main/contracts/ERC721WithPermit.sol#L116) uses OZ's [`SignatureChecker` that in turn implements EIP-1271](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/SignatureChecker.sol#L36)
- Unclear to me: Whether or not the authors of the EIP document want to further mention 1271. Anyways, at least I think it should be mentioned in the header as required.

EIP-2098:
- document mentions: "sig is a valid secp256k1 or [EIP-2098](https://eips.ethereum.org/EIPS/eip-2098) signature from owner of the tokenId:"

### CODE REVIEW

**Comment by Pandapip1** [comment]
@MicahZoltu @lightclient could one of you manually merge this?

**Comment by MicahZoltu** [comment]
We need to get approval from one of the authors before merging.  The author may prefer to remove the reference to 2098 to resolve that one, and they may not want to require 1271 for some reason.  We should not let 4494 go to review without resolving the 2098 issue though.

**Comment by TimDaub** [comment]
@dievardump @wschwab ☝🏼

**Comment by wschwab** [comment]
I don't see why 1271 or 2098 should be required. We have designed the EIP to be compatible with them, but at no point in the implementation are they required. Feel free to change my mind.

**Comment by Pandapip1** [comment]
> Feel free to change my mind.

It should be explicitly stated that they may be supported, then, because EIP-1271 is assumed to not be supported.

**Comment by TimDaub** [comment]
> I don't see why 1271

fair.

> 2098 should be required

You mention EIP-2098 as also @dievardump points out, and it is how you format the signature as `bytes`, and in that case, IMO, it should be added to the requirements.

**Comment by TimDaub** [comment]
> we point it out, but we never say that a 4494-compliant implementation MUST support either, imho a requirement should be something required, not something that was merely accommodated

fine. defining it like this can be a pro and a con.

Pro: if a better compact signature scheme arrives it may be usable

con: being loose can lead to undefined behavior.

I tend to prefer being very explicit and specific but it's ur guy's call, feel free to close

File: EIPS/eip-4494.md
Comment (xinbenlv) [inline EIPS/eip-4494.md:4]: @Pandapip1 it seems a false positive warning for EIPW? My understanding is that preamble can't add Markdown Links.

**Review by xinbenlv — APPROVED** [review]

File: EIPS/eip-4494.md
Comment (Pandapip1) [inline EIPS/eip-4494.md:4]:
```suggestion
description: EIP-712-singed approvals for EIP-721 NFTs
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
